perma_web/Dockerfile

```diff
@@ -36,10 +36,8 @@ RUN apt-get update \
         libxml2-dev \
         libxslt-dev
 
-# pin node version -- see https://github.com/nodesource/distributions/issues/33
-RUN curl -o nodejs.deb https://deb.nodesource.com/node_14.x/pool/main/n/nodejs/nodejs_14.19.3-deb-1nodesource1_$(dpkg --print-architecture).deb \
-    && dpkg -i ./nodejs.deb \
-    && rm nodejs.deb
+# node.js (needs to be pinned?)
```

Comment:
The mechanism for installing node has changed; I believe it's no longer possible to pin to a specific .deb file exactly like we used to; the new mechanism is at https://github.com/nodesource/distributions/issues/33#issuecomment-1698870039

Reply:
> ideally to the version in prod
I note that we do not run nodejs in prod for Perma: it only ever runs inside the docker container

Without pinning here, I think the biggest possible consequences are, a discrepancy between different developers' environments, and an accidental upgrade to a new minor release breaking something that was previously working.

Reply:
I don't feel strongly -- I just feel like I've been punished every time I let versions slip between different dev envs or between dev and CI, including I swear I remember adding node pinning to one of our projects after it wasted some time debugging something or other, so wanted to pass that on.

Reply:
@bensteinberg @jcushman I believe this would work, should I push and see if that works?
```bash 
# node.js 
RUN curl -fsSL https://deb.nodesource.com/setup_20.x | bash - && apt-get install -y nodejs=20.12.0-1nodesource1
```

Reply:
Perfect, I was about to make this exact suggestion, and point out that I _just_ installed Debian updates for node on a few machines.

Reply:
@matteocargnelutti is there a build relic to remove after the fact, like the `rm` move we had before? asking just because this image is so big

Reply:
@rebeccacremona As in: it got much bigger with that change? I am not sure 👀 !